File: 프로그래머스/3/42884. 단속카메라/단속카메라.js
function solution(routes) {
  // 1) 진출 지점(끝점) 기준 오름차순 정렬
  routes.sort((a, b) => a[1] - b[1]);

  let cameras = 0;
  let camPos = -30001; // 범위(-30000~30000) 밖에서 시작

  // 2) 그리디: 아직 안 걸린 구간을 만나면 그 구간의 끝점에 카메라 설치
  for (const [start, end] of routes) {
    if (camPos < start) {
      cameras++;
      camPos = end;
    }
  }

  return cameras;
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.py
def solution(routes):
    # 진출 지점 기준으로 정렬
    routes.sort(key=lambda x: x[1])

    cameras = 0
    cam_pos = -30001  # 가능한 범위 밖에서 시작

    for start, end in routes:
        # 현재 카메라 위치가 이 차량 구간에 없으면 새로 설치
        if cam_pos < start:
            cameras += 1
            cam_pos = end  # 이 차량의 진출 지점에 설치

    return cameras